Added component example for single page viewer

pixelexel opened the pull request:
This pull request is in response to issue #8951 and adds a component example for single page viewer in the `example/components` folder.

Fixes #8951.

Snuffleupagus commented on examples/components/singlepageviewer.html:
```diff
@@ -0,0 +1,51 @@
+  <link rel="stylesheet" href="../../node_modules/pdfjs-dist/web/pdf_viewer.css">
+
+  <script src="../../node_modules/pdfjs-dist/build/pdf.js"></script>
+  <script src="../../node_modules/pdfjs-dist/web/pdf_single_page_viewer.js"></script>
```
This file doesn't exist, which explains why the example currently doesn't work.
Please look at the contents of the folder in question, and you'll see that the path must be `../../node_modules/pdfjs-dist/web/pdf_viewer.js`.

pixelexel replied:
Also, should the condition check for `PDFJS.PDFViewer` instead of `PDFJS.PDFSinglePageViewer` here: https://github.com/pixelexel/pdf.js/blob/b24cdc08e24fc1e99c6a56eab807b80c15a72965/examples/components/singlepageviewer.js#L18
like in the other two examples?

Snuffleupagus commented on examples/components/singlepageviewer.js:
```diff
@@ -0,0 +1,68 @@
+
+// (Optionally) enable find controller.
+var pdfFindController = new PDFJS.PDFFindController({
+  pdfSinglePageViewer: pdfSinglePageViewer
```
I don't think this would work since, as can be seen in https://github.com/mozilla/pdf.js/blob/master/web/pdf_find_controller.js#L49, the parameter is named `pdfViewer`. Hence this lince needs to read `pdfViewer: pdfSinglePageViewer,`.

Snuffleupagus commented on examples/components/singlepageviewer.js:
```diff
@@ -0,0 +1,68 @@
+pdfSinglePageViewer.setFindController(pdfFindController);
+
+container.addEventListener('pagesinit', function () {
+    // We can use pdfSinglePageViewer now, e.g. let's change default scale.
```
Nit: Please remove two spaces of indentation here, such that the comment lines up correctly with the line below.

Snuffleupagus reviewed (changes requested):
Unfortunately this doesn't (currently) work, please see the comments below.
Please remember to test locally, to make sure that the new example works, before submitting a new version of the patch (and please squash the commits).

pixelexel commented on package.json:
```diff
@@ -43,5 +43,8 @@
     "type": "git",
     "url": "git://github.com/mozilla/pdf.js.git"
   },
-  "license": "Apache-2.0"
```
I don't understand what's happening here.
I didn't change the package.json file and still it shows as edited and the build on travis-ci seems to have failed.

pixelexel committed:
Added component example for single page viewer

Checking for PDFJS.PDFSinglePageViewer instead

Added component example for single page viewer

Snuffleupagus commented on examples/components/singlepageviewer.js:
```diff
@@ -0,0 +1,68 @@
+
+'use strict';
+
+if (!PDFJS.PDFViewer || !PDFJS.getDocument) {
```
Why are you checking for `PDFViewer` when you're not using that below?
This line *should* read if (!PDFJS.PDFSinglePageViewer|| !PDFJS.getDocument) {` instead.

Snuffleupagus reviewed (approved):
Looks good, thank you for the patch!

pixelexel commented:
Honestly, thank you for putting up with me. Looking back, I made so many stupid and obvious mistakes while working on this but you always helped me along. 
Thanks a lot for your time and I look forward to learning more and contributing in a much better way next time!

timvandermeij commented:
Thank you for your contribution! No worries about that; reviews are there to improve everyone's work and especially the first contribution to a project is always aimed at getting a feeling of the entire workflow, which always takes some time to get used to. You're always welcome to contribute and ask questions!
